OnStartup.cpp

```
@@ -0,0 +1,101 @@
+#include <iostream>
+#include <windows.h>
+
+constexpr int VALID_EXIT_CODE{ 0 };
+constexpr int INVALID_EXIT_CODE{ 1 };
+constexpr int ONE_HOUR{ 1000 * 60 * 60 };
+constexpr LPCWSTR REGISTRY_STARTUP_DIR{ L"Software\\Microsoft\\Windows\\CurrentVersion\\Run" };
+constexpr LPCWSTR REGISTRY_STARTUP_VALUE{ L"sleepForAnHour" };
+
+using std::cout;
+using std::endl;
+
+/*
+* Creates a mutex for this program
+* 
+* @param[OUT] mutex The mutex created by the function
+* 
+* @return An exit code represanting if there was an error or not
+*/
+int createCustomMutex(HANDLE* mutex) {
+	*mutex = CreateMutexW(
```

Why not create the mutex here and pass it outside of function? Instead of returning an int, throw exception on failure

---

I am not sure i understand the first sentence